// my cleverest game proto
syntax = "proto3";

package  game;

service Game {
  rpc connection (User) returns (Status);
  rpc disconnection (User) returns (Status);
  rpc answer_question (AnswerRequest) returns (AnswerResponse);
  rpc get_result_table (Empty) returns (TotalRating);
  rpc get_question_stream (Empty) returns (stream Question);
}

message User{
  string username = 1;
}

message Status{
  bool result = 1;
}

message Empty {}

message TotalRating{
  string total_rating = 1;
}

message Question{
  int32 number_question = 1;
  string text_question = 2;
  repeated string answers = 3;
  int32 correct_answer = 4; // number of answer in list
}


message AnswerRequest{
  int32 number_question = 1;
  string username = 2;
  int32 user_answer = 3;
}

message AnswerResponse{
  bool answer_score = 1;
  int32 rating = 2;
}